### PR TITLE
Only show the UserManagement tab in the sidebar if the sub menus are allowed for that cluster

### LINKED ISSUE #2615: access controls should be hidden when inaccessible
**Describe the bug**
If the user does not have permission to get/list service accounts, roles, and/or role bindings, these buttons are still showing up under "Access Control". If you click on them, it says "Failed to load items."

**To Reproduce**
1. Set up a user with limited permissions with no access to service accounts, roles, or role bindings.
2. Authenticate as that user and open up the cluster in Lens.
3. Open up "Access Controls" on the lefthand side and click on "Service Accounts", "Role Bindings", or "Roles".

**Expected behavior**
These buttons should be hidden if the user doesn't have permission, just like other resources are hidden. If nothing under "Access Controls" is accessible then the whole thing should be hidden.

**Screenshots**
image

**Environment (please complete the following information):**
- Lens Version: 4.2.2
- OS: macos 10.15.7
- Installation method (e.g. snap or AppImage in Linux): n/a

### PR DESCRIPTION
**Pull request by Nokel81**
breaking out part of #2003 

Signed-off-by: Sebastian Malton <sebastian@malton.name>

### CODE REVIEW

File: src/common/rbac.ts [hunk 1]
```diff
@@ -4,42 +4,54 @@ export type KubeResource =
   "namespaces" | "nodes" | "events" | "resourcequotas" | "services" | "limitranges" |
   "secrets" | "configmaps" | "ingresses" | "networkpolicies" | "persistentvolumeclaims" | "persistentvolumes" | "storageclasses" |
   "pods" | "daemonsets" | "deployments" | "statefulsets" | "replicasets" | "jobs" | "cronjobs" |
-  "endpoints" | "customresourcedefinitions" | "horizontalpodautoscalers" | "podsecuritypolicies" | "poddisruptionbudgets";
+  "endpoints" | "customresourcedefinitions" | "horizontalpodautoscalers" | "podsecuritypolicies" | "poddisruptionbudgets" |
+  "role" | "clusterrole" | "rolebinding" | "clusterrolebinding" | "serviceaccount";
```
Comment (jim-docker) [inline src/common/rbac.ts]: shouldn't these be plural like all the others? Or does this come from the kube api?

File: src/common/rbac.ts [hunk 2]
```diff
@@ -4,42 +4,54 @@ export type KubeResource =
...
+  "persistentvolumeclaims": { kind: "PersistentVolumeClaim" },
+  "pods": { kind: "Pod" },
+  "poddisruptionbudgets": { kind: "PodDisruptionBudget", group: "policy" },
+  "podsecuritypolicies": { kind: "PodSecurityPolicy" },
+  "resourcequotas": { kind: "ResourceQuota" },
+  "replicasets": { kind: "ReplicaSet", group: "apps" },
+  "role": { kind: "Role", group: "rbac.authorization.k8s.io" },
+  "rolebinding": { kind: "RoleBinding", group: "rbac.authorization.k8s.io" },
+  "secrets": { kind: "Secret" },
+  "serviceaccount": { kind: "ServicAccount", group: "core" },
```
Comment (jim-docker) [inline src/common/rbac.ts]:
```suggestion
  "serviceaccount": { kind: "ServiceAccount", group: "core" },
```

File: src/renderer/components/+user-management/user-management.tsx [hunk 3]
```diff
@@ -13,29 +13,37 @@ import { isAllowedResource } from "../../../common/rbac";
...
         component: RoleBindings,
         url: roleBindingsURL({ query }),
         routePath: roleBindingsRoute.path.toString(),
-      },
-      {
+      });
+    }
+
+    if (isAllowedResource("role") || isAllowedResource("clusterrole")) {
+      // TODO: seperate out these two pages
```
Comment (ixrock) [inline src/renderer/components/+user-management/user-management.tsx:39]: Why those pages should be separated?

**Review by ixrock — DISMISSED** [review]

File: src/renderer/components/+user-management/user-management.tsx (hunk 3 above)
Comment (Nokel81) [reply src/renderer/components/+user-management/user-management.tsx:39]: It relates to #2199. Namely, they are different resources with slightly different semantics so they should be on different pages.

**Review by jim-docker — APPROVED** [review]

**Review by nevalla — APPROVED** [review]
LGTM

**Pull request merged**